### PR TITLE
ProxyObject: support of pickle

### PR DESCRIPTION
**Pull request by madsbk**
This PR implements `ProxyObject.__reduce__()` in order to support pickle.

### CODE REVIEW

File: dask_cuda/proxy_object.py [hunk 1]
```diff
@@ -155,15 +156,31 @@ def __init__(
...
+            "fixed_attr",
+            "type_serialized",
+            "typename",
+            "is_cuda_object",
+            "subclass",
+            "serializers",
+        ]
+        if include_obj:
+            args.insert(0, "obj")
+        return OrderedDict([(a, self._obj_pxy[a]) for a in args])
```
Comment (madsbk) [inline dask_cuda/proxy_object.py:183]: Notice, even though `dict` is order as of Python 3.7, I think it make sense to use `OrderedDict` to make it explicit.

**Review by pentschev — APPROVED** [review]
I have a minor suggestion but other than that it looks good to me. In fact, that change is just a suggestion to make code a bit more understandable (at least I think it does), but feel free to ignore that either.

File: dask_cuda/proxy_object.py [hunk 2]
```diff
@@ -155,15 +156,31 @@ def __init__(
...
+        args = [
+            "fixed_attr",
+            "type_serialized",
+            "typename",
+            "is_cuda_object",
+            "subclass",
+            "serializers",
+        ]
+        if include_obj:
+            args.insert(0, "obj")
```
[the remark this answers is not in the document]
Comment (madsbk) [reply dask_cuda/proxy_object.py:182]: Good point fixed!

File: dask_cuda/device_host_file.py [hunk 3]
```diff
@@ -56,7 +56,7 @@ def device_deserialize(header, frames):
 
 @nvtx_annotate("SPILL_D2H", color="red", domain="dask_cuda")
 def device_to_host(obj: object) -> DeviceSerialized:
-    header, frames = serialize(obj, serializers=["dask", "pickle"])
+    header, frames = serialize(obj, serializers=["dask", "pickle"], on_error="raise")
```
Comment (jakirkham) [inline dask_cuda/device_host_file.py:59]: Thanks for catching this 😄

**Pull request merged**